### ISSUE TITLE
How to use specific IP address for an instance?

### ISSUE THREAD

**Issue by Games4k** [retspen/webvirtcloud#349]
Hi there. After a long time, I've finally figured out how to setup webvirtcloud. The issue always was that you did not specify we need to add www-data to a specific group.

Now, I have a problem. I don't know how to allocate a specific IP address to a specific instance. I would like to request help in that segment, as there are 0 guides on that part. I am able to create an instance, install OS, and connect to it via "Console access", but for example, if I host a website on that machine, I can't access it it does not have an IP address.

Thanks!

**Comment by catborise** [comment]
how do you access your host?
is there two ip on the host interface? (private ip, public ip?)

**Comment by Games4k** [comment]
> how do you access your host?
> is there two ip on the host interface? (private ip, public ip?)

I do have access to the` IP | Subnet | rDNS | Gateway | Netmask` of each IP address. Sadly, I do not have access to a private IP address, there is only a public one. My current provider is https://zare.com btw.

**Comment by catborise** [comment]
there are some options related with networking. you should focus on that.
according to your purpose you can use bridges or nats or routed networks or macvtaps. 
what is your purpose?
after creating an instances what will you do these instances.

**Comment by Games4k** [comment]
I'm planning on selling those as VPSes. SolusVM is taking a significant part of revenue, so I do not really think it is great for such things, while this, as a free alternative is great. Proxmox is good, but it's a standalone OS pretty much and I require something that would work with ubuntu. 

Overall, WebVirtCloud is the best option, it has a great interface, a lot of features, and all that for free, which would make prices of my VPSes significantly lower than other competitors.

**Comment by purerosefallen** [comment]
You have to focus on iptables. Disable their networks when using wrong IP addresses.
~~Is WebVirtCloud WHMCS addon coming?~~

**Comment by Games4k** [comment]
@purerosefallen were you able to achieve that effect. If so, please contact me on discord OpenSource#3310.

@catborise do you have any guides or ideas on what I have to do, even on external websites? I'd really like to start using webvirtcloud, but no matter what I try, I am not able to bind a specific IP address to a specific server.

**Comment by catborise** [comment]
it is not related with virtual machines. it is up to your network configuration. may be you can use a reverse proxy / load balancer etc , if you have only one real ip.

**Comment by Games4k** [comment]
No, I’ve purchased an additional real IP for the VPS, just don’t know how to assign it so other people can use that IP to connect to that VPS

**Comment by Games4k** [comment]
No, I’ve purchased an additional real IP for the VPS, just don’t know how to assign it so other people can use that IP to connect to that VPS @catborise

**Comment by Games4k** [comment]
Here is an example of what I want to achieve:

I have a host dedicated server on 1.0.0.0 (as an example).

I've created a virtual private server and purchased an additional IP for it from my hosting provider

Now, I need to assign that IP to that VPS so users can access it through that IP

**Comment by catborise** [comment]
https://superuser.com/questions/1284491/one-public-ip-and-multiple-vps-servers
may be you should install a gateway like pfsense then route requests to your instances. 
this requires extra manual works. as i said before you should use webmin/virtualmin.

**Comment by purerosefallen** [comment]
NAT VPS?
